### DISCUSSION TITLE
Participating in LXQt

### DISCUSSION THREAD

**Discussion by francescmaestre** [lxqt/lxqt#2222] [General]
Hi all,

I found this repo and I'm interested in giving a hand with the work for some of the apps (specially QTermWidgeT) if you want!

(@tsujan I've [already added a PR](https://github.com/lxqt/qtermwidget/pull/469), but after checking the code more in-depth I have a lot of suggestions for improvements, code modernizations, etc.)

I have a lot of experience with Qt (since 4.2 and even earlier) and I think I could bring some help here. If you want to take a look at some of my code, please check the [GitQlient](https://github.com/francescmm/GitQlient) repo.

I'm not sure how to start with the contribution process and all of that, so just tell me! :)

**Comment by tsujan** [comment]
Welcome to LXQt and thank you very much for your contribution!

> I'm not sure how to start with the contribution process

You have already started it :)

The contributor makes a PR, which will be reviewed by a developer who masters the code — in this case, @yan12125. The process of reviewing depends on the PR. Usually, the reviewer asks for this or that change; discussions may also start.

All LXQt members are volunteers who have dedicated parts of their free times to LXQt. It may take a while before a review gets started. We know that @yan12125 has been very busy recently. So, please be patient! Your PR won't be ignored.

**Reply by tsujan** [reply]
> but after checking the code more in-depth I have a lot of suggestions for improvements, code modernizations, etc.

That's very good. You could make PRs, but please make one PR for each change.

Each PR should be about one subject (fix, change or feature). Only in that way, we could keep track of changes and find the suspicious ones in case a regression happens or a new bug shows up.

**Comment by francescmaestre** [comment]
Then I'll start with that!

Another thing I was thinking on was joining as a volunteer at LXQt in the near future if I see I'm having time, and once I've been contributing for a bit :)

See you in the PRs! :smile:

**Reply by tsujan** [reply]
> Another thing I was thinking on was joining as a volunteer at LXQt in the near future

That would be very kind of you. LXQt definitely needs more good developers.

After some of their good works are merged and when they show willingness to do more, contributors are invited by LXQt.

**Comment by yan12125** [comment]
Thanks a lot for the contributions! I left some first-glance comments, and in-depth review will follow when I find time.
